modules/core/src/main/java/org/apache/synapse/config/xml/CallMediatorSerializer.java

```diff
@@ -46,6 +46,9 @@ public OMElement serializeSpecificMediator(Mediator m) {
         if (activeEndpoint != null) {
             call.addChild(EndpointSerializer.getElementFromEndpoint(activeEndpoint));
         }
+        if (mediator.getBlocking() == true) {
+            call.addAttribute(fac.createOMAttribute("blocking", nullNS, Boolean.toString(mediator.getBlocking())));
```

Comment:
Yes,For the boolean,no need the additional operation.the suggested change is fixed in the latest code